### PR TITLE
py-xdot requires explicit dependency on harfbuzz

### PR DESCRIPTION
**Pull request by lee218llnl**
This PR fixes a runtime error with xdot:

> ImportError: Typelib file for namespace 'HarfBuzz', version '0.0' not found

### CODE REVIEW

**Comment by lee218llnl** [comment]
@adamjstewart how does this look to you? Do you have any questions or change requests?

File: var/spack/repos/builtin/packages/py-xdot/package.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class PyXdot(PythonPackage):
     depends_on('py-pygobject', type=('build', 'run'))
     depends_on('py-pycairo', type=('build', 'run'))
     depends_on('pango', type=('build', 'run'))
+    depends_on('harfbuzz', type=('build', 'run'))
     depends_on('atk', type=('build', 'run'))
     depends_on('gdk-pixbuf', type=('build', 'run'))
     depends_on('gtkplus', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xdot/package.py:33]: Should these have a link dep? If the compiler uses them, then we should add a link dep. If it's only used post-install, then we can keep them build/run.

File: var/spack/repos/builtin/packages/py-xdot/package.py [hunk 2]
```diff
@@ -35,11 +36,12 @@ class PyXdot(PythonPackage):
...
     def post_install(self):
         spec = self.spec
-        repo_paths = '%s:%s:%s:%s' % (
+        repo_paths = '%s:%s:%s:%s:%s' % (
             join_path(spec['pango'].prefix.lib, 'girepository-1.0'),
             join_path(spec['atk'].prefix.lib, 'girepository-1.0'),
             join_path(spec['gdk-pixbuf'].prefix.lib, 'girepository-1.0'),
-            join_path(spec['gtkplus'].prefix.lib, 'girepository-1.0'))
+            join_path(spec['gtkplus'].prefix.lib, 'girepository-1.0'),
+            join_path(spec['harfbuzz'].prefix.lib, 'girepository-1.0'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xdot/package.py:44]: Should we be using `spec['foo'].libs.directories[0]` to properly handle cases where the libraries are in `lib64`, not `lib`?

**Review by adamjstewart — COMMENTED** [review]
Thanks for the ping, this somehow fell off my radar.

File: var/spack/repos/builtin/packages/py-xdot/package.py (hunk 1 above)
Comment (lee218llnl) [reply var/spack/repos/builtin/packages/py-xdot/package.py:33]: xdot itself is pure python so it doesn't require linking to harfbuzz

File: var/spack/repos/builtin/packages/py-xdot/package.py (hunk 2 above)
Comment (lee218llnl) [reply var/spack/repos/builtin/packages/py-xdot/package.py:44]: this is for .typelib files, not .so files, so I don't think they will ever be installed in lib64

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-xdot/package.py:44]: What is a `.typelib` file?

Comment (lee218llnl) [reply var/spack/repos/builtin/packages/py-xdot/package.py:44]: I'm no expert on this, so I will just refer to https://gi.readthedocs.io/en/latest/

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-xdot/package.py:44]: Is this something that xdot creates and writes to these installation prefixes?

Comment (lee218llnl) [reply var/spack/repos/builtin/packages/py-xdot/package.py:44]: xdot does not create them, they are created by the dependent packages (harfbuzz, pango, etc.)

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Pull request merged**